client/main.cpp

```diff
@@ -122,10 +128,26 @@ int main(int argc, char** argv)
 
     parser::KlassRegister klass_register;
     parser::ProtocolReader reader(&klass_register, std::move(stream), true);
-    client::ChromeTraceListener chrome_listener(std::move(tracepoint_map));
     std::string out_file = create_output_path(output_path.c_str());
-    chrome_listener.init(out_file);
-    reader.register_events_listener([&chrome_listener] (const parser::Event& event) { chrome_listener.process_event(event); });
+
+
+    if (format == "chrome")
+    {
+        client::ChromeTraceListener chrome_listener(std::move(tracepoint_map));
+        chrome_listener.init(out_file);
+        reader.register_events_listener([&chrome_listener] (const parser::Event& event) { chrome_listener.process_event(event); });
+    }
+    else if (format == "callgrind")
+    {
+        client::CallgrindListener callgrind_listener(std::move(tracepoint_map));
+        callgrind_listener.init(out_file);
+        reader.register_events_listener([&callgrind_listener] (const parser::Event& event) { callgrind_listener.process_event(event); });
+        std::cout << "Callgrind format is not implemented yet" << std::endl;
+    }
+    else
+    {
+        std::cerr << "Unknown format: " << format << ". Supported formats are chrome-tracing and callgrind" << std::endl;
```

Comment:
Looks like you repeat "chrome-tracing" and "callgrind" 3 times (here, in the documentation, and in `if` statements. Could we have only one place where we define all possible options (like a map)?